lib/rules/await-fire-event.ts

```diff
@@ -1,51 +1,106 @@
-export type MessageIds = 'awaitFireEvent';
+export type MessageIds = 'awaitFireEvent' | 'fireEventWrapper';
 type Options = [];
-export default ESLintUtils.RuleCreator(getDocsUrl)<Options, MessageIds>({
+
+export default createTestingLibraryRule<Options, MessageIds>({
   name: RULE_NAME,
   meta: {
     type: 'problem',
     docs: {
-      description: 'Enforce async fire event methods to be awaited',
+      description: 'Enforce promises from fire event methods to be handled',
       category: 'Best Practices',
       recommended: false,
     },
     messages: {
-      awaitFireEvent: 'async `fireEvent.{{ methodName }}` must be awaited',
+      awaitFireEvent:
+        'Promise returned from `fireEvent.{{ methodName }}` must be handled',
+      fireEventWrapper:
+        'Promise returned from `fireEvent.{{ wrapperName }}` wrapper over fire event method must be handled',
     },
     fixable: null,
     schema: [],
   },
   defaultOptions: [],
 
-  create: function (context) {
+  create: function (context, _, helpers) {
+    const functionWrappersNames: string[] = [];
+
+    function reportUnhandledNode(
+      node: TSESTree.Identifier,
+      closestCallExpressionNode: TSESTree.CallExpression,
+      messageId: MessageIds = 'awaitFireEvent'
+    ): void {
+      if (!isPromiseHandled(node)) {
+        context.report({
+          node: closestCallExpressionNode.callee,
+          messageId,
+          data: { name: node.name },
+        });
+      }
+    }
+
+    function detectFireEventMethodWrapper(node: TSESTree.Identifier): void {
+      const innerFunction = getInnermostReturningFunction(context, node);
+
+      if (innerFunction) {
+        functionWrappersNames.push(getFunctionName(innerFunction));
+      }
+    }
+
     return {
-      'CallExpression > MemberExpression > Identifier[name=fireEvent]'(
-        node: TSESTree.Identifier
-      ) {
-        const memberExpression = node.parent as TSESTree.MemberExpression;
-        const fireEventMethodNode = memberExpression.property;
-
-        if (
-          ASTUtils.isIdentifier(fireEventMethodNode) &&
-          !isAwaited(node.parent.parent.parent) &&
-          !hasChainedThen(fireEventMethodNode.parent)
-        ) {
-          context.report({
-            node: fireEventMethodNode,
-            messageId: 'awaitFireEvent',
-            data: {
-              methodName: fireEventMethodNode.name,
-            },
-          });
+      'CallExpression Identifier'(node: TSESTree.Identifier) {
+        if (helpers.isFireEventMethod(node)) {
+          detectFireEventMethodWrapper(node);
+
+          const closestCallExpression = findClosestCallExpressionNode(
+            node,
+            true
+          );
+
+          if (!closestCallExpression) {
+            return;
+          }
+
+          const references = getVariableReferences(
+            context,
+            closestCallExpression.parent
+          );
+
+          if (!references || references.length === 0) {
```

Comment:
Nit, I believe `getVariableReferences` will return an empty array when there are no references

```suggestion
          if (references.length === 0) {
```